pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -83,11 +94,21 @@ func (r *reconciler) ReconcileActiveGateCommunicationHosts(ctx context.Context,
 
 	activeGateEndpoints := activegate.GetEndpointsAsCommunicationHosts(dynakube)
 
+	const conditionComponent = "ActiveGate"
+
 	err := r.reconcileCommunicationHostsForComponent(ctx, activeGateEndpoints, ActiveGateComponent)
 	if err != nil {
+		setServiceEntryFailedConditionForComponent(dynakube.Conditions(), conditionComponent, err)
+
 		return err
 	}
 
+	if len(activeGateEndpoints) > 0 {
+		setServiceEntryUpdatedConditionForComponent(dynakube.Conditions(), conditionComponent)
+	} else {
+		meta.RemoveStatusCondition(dynakube.Conditions(), getConditionTypeName(conditionComponent))
```

Comment:
one of the Acceptance Criteria in the linked ticket:
- `condition and istio resources are cleaned up, when not necessary`

you do a clean up here for the condition (and in line 107, the istio stuff is also cleaned up)

but in line 91, we do not clean up, if the activeGate was "turned off", and we should

(also the log line in line 88 is incorrect, but that is a mistake done by someone else 😅 )

the clean-up should also be double-checked in the `ReconcileCodeModuleCommunicationHosts` func, but maybe it just works 🤔

Reply:
fixed in 2d824c6d

I think that this is no issue for the CodeModules as the conditions are 
* set to failed if there were errors
* set to updated if there are CommunicationHosts where the ServiceEntries were updated
* removed if there are none

Reply:
the `if !dynakube.NeedsActiveGate` check ([line 93-97](https://github.com/Dynatrace/dynatrace-operator/pull/3198/files#diff-6dcf84623910535feca0c7dc62e7a35bd7706001d348bc3c88ba5f69893b100cL80-L82)) will stop the cleanup of the istio resources we created for the ActiveGate, as the clean up happens deeper in the `r.reconcileCommunicationHostsForComponent` :
- [here](https://github.com/Dynatrace/dynatrace-operator/blob/e93ae0ad8f1a0552cfce5ff7ee5818809fe3e6d8/pkg/controllers/dynakube/istio/reconciler.go#L150-L157) and [here](https://github.com/Dynatrace/dynatrace-operator/blob/e93ae0ad8f1a0552cfce5ff7ee5818809fe3e6d8/pkg/controllers/dynakube/istio/reconciler.go#L184-L194)

we don't have a similar check for the CodeModules, so those get cleaned up (probably, only looked at code, didn't run tests)

the most "efficient" way to handle the creation/update/delete with the help of the condition: (we already do this for 1 or 2 secrets)
1. `if needed && conditionNotSet` => Create resources and set condition
2. `if needed && conditionPresent && conditionUpToDate` => do nothing
3. `if needed && conditionPresent && conditionNOTUpToDate` => update resources and condtion
4. `if notNeeded && conditionPresent` => cleanup resources and condition
5. `if notNeeded && conditionNotPresent` => do nothing

(it looks something like [this](https://github.com/Dynatrace/dynatrace-operator/blob/290c4dac9064a4c97be150f373c8e39c7dcd6aed/pkg/controllers/dynakube/connectioninfo/activegate/reconciler.go#L43-L69), its not just a bunch of `ifs`, more of a flow)

Reply:
fixed in ecac0abd